docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,368 @@
+
+The Auth Service supports the following backends for cluster state and audit
+events:
+
+- Amazon DynamoDB
+- Google Cloud Firestore
+
+For Amazon DynamoDB and Google Cloud Firestore, your Teleport configuration (which
+we will describe in more detail in the [Configuration](#configuration) section)
+names a table or collection where Teleport stores cluster state and audit
+events. 
+
+The Teleport Auth Service manages the creation of any required DynamoDB tables
+or Firestore collections itself, and does not require them to exist in advance.
+
+The Auth Service can also store cluster state in self-hosted etcd deployments.
+In this case, Teleport uses namespaces within item keys to identify cluster
+state data.
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution (e.g., AWS or Google Cloud IAM), your
+Auth Service instances need permissions to read from and write to your chosen
+key/value store, as well as to create tables and collections (if your key/value
+store supports them).
+
+</Admonition>
+
+## Object storage service
+
+High-availability Teleport deployments use an object storage service for
+persisting session recordings. The Teleport Auth Service supports two object
+storage services:
+
+- Amazon S3 
+- Google Cloud Storage
+
+In your Teleport configuration (described in the [Configuration](#configuration)
+section), you must name a Google Cloud Storage or Amazon S3 bucket to use for
+managing session recordings. The Teleport Auth Service creates this bucket, so
+to prevent unexpected behavior, you should not create it in advance. 
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution, your Auth Service instances need
+permissions to create, read from, and write to buckets.
+
+</Admonition>
+
+## TLS credential provisioning 
+
+High-availability Teleport deployments require an automated system to fetch TLS
+credentials from a certificate authority like Let's Encrypt, AWS Certificate
+Manager, Digicert, or a trusted internal authority. The system must then
+provision Teleport Proxy Service instances with these credentials and renew them
+periodically. 
+
+If you are running a single instance of the Teleport Auth Service and Proxy
+Service, you can configure this instance to fetch credentials for itself from
+Let's Encrypt using the [ACME ALPN-01
+challenge](https://letsencrypt.org/docs/challenge-types/#tls-alpn-01), where
+Teleport demonstrates that it controls the ALPN server at the HTTPS address of
+your Teleport Proxy Service.
+
+For high-availability deployments that use Let's Encrypt to supply TLS
+credentials to Teleport instances running behind a load balancer, you will need
+to use the [ACME
+DNS-01](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge)
+challenge to demonstrate domain name ownership to Let's Encrypt. In this
+challenge, your TLS credential provisioning system creates a DNS TXT record with
+a value expected by Let's Encrypt.
```

Comment:
A DNS-01 challenge must also be used to get a wildcard cert from Let's Encrypt (which may be useful in the context of Application Access)

Reply:
Testing a single-instance Teleport deployment with one application registered (this guide: https://goteleport.com/docs/deploy-a-cluster/open-source/), it looks like Teleport requests a certificate for each registered application's subdomain individually from Let's Encrypt, rather than directly requesting a cert for a wildcard domain. Are there instances where Teleport does request a cert for a wildcard domain?

Reply:
Yes, you're absolutely right that Teleport's built-in ACME will get certs for each individual application and maintain those separately. Teleport itself never requests wildcard certs, so it's probably not relevant here!